compiler/include/resolution.h

```diff
@@ -29,7 +29,9 @@ class CallInfo;
 
 extern SymbolMap paramMap;
 extern Vec<CallExpr*> callStack;
-extern Map<Type*,FnSymbol*> autoCopyMap; // type to chpl__autoCopy function
+bool hasAutoCopyForType(Type* type);
```

Comment:
I agree with Michael.  The NULL indicator is poor coding practice in my mind on both our homegrown Map and the C++ map class.  The boolean indicator is a much more principled strategy and one that is used in higher-level languages.